Add docs about not running e2e testing m directly (#2629)
operator-framework/operator-sdk: doc/test-framework/writing-e2e-tests.md

[Framework][framework-link] contains all global variables, such as the kubeconfig, kubeclient,
scheme, and dynamic client (provided via the controller-runtime project).

> It is initialized by the MainEntry function and can be used anywhere in the tests. As several

Reviewer: ```suggestion
It is initialized by the MainEntry function and can be used anywhere in the tests.
```

> required arguments are initialized and added by MainEntry, do not attempt to run the testing M

Reviewer: ```suggestion

**Note:** several required arguments are initialized and added by `MainEntry()`. Do not attempt to
use `testing.M` directly.
```

> directly.

Reviewer: ```suggestion
```
